amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -8,492 +8,967 @@ input AMPLIFY {
   createUniqueUser(input: CreateUniqueUserInput): User
     @function(name: "createUniqueUser")
+  """Create a unique Colony within the Colony Network. Use this instead of the automatically generate `createColony` mutation"""
   createUniqueColony(input: CreateUniqueColonyInput): Colony
     @function(name: "createUniqueColony")
+  """Updates the latest available version of a Colony or an extension"""
   setCurrentVersion(input: SetCurrentVersionInput): Boolean
     @function(name: "setCurrentVersion")
+  """
+  Update an extension's details for a specific Colony.
+
+  The extension hash is generated like so: `keccak256(toUtf8Bytes(extensionName))`, where `extensionName` is the name of the extension contract file in the Colony Network (e.g. `VotingReputation`)
+  """
   updateExtensionByColonyAndHash(
     input: UpdateExtensionByColonyAndHashInput
   ): ColonyExtension @function(name: "updateExtensionByColonyAndHash")
 }
 
+"""
+Represents a user's profile within the Colony Network.
+"""
 type Profile @model {
+  """Unique identifier for the user's profile."""
   id: ID!
+  """URL of the user's avatar image."""
   avatar: String
+  """URL of the user's thumbnail image."""
   thumbnail: String
+  """Display name of the user."""
   displayName: String
+  """User's bio information."""
   bio: String
+  """User's location information."""
   location: String
+  """URL of the user's website."""
   website: AWSURL
+  """User's email address."""
   email: AWSEmail @index(name: "byEmail", queryField: "getProfileByEmail")
+  """Metadata associated with the user's profile."""
   meta: ProfileMetadata
 }
 
+"""
+Represents the status of a Colony's native token.
+
+Colonies can have different types of native tokens in various modes. Here we define some important properties that the dApp uses to enable or disable certain features or views. This is set when a Colony is created and can be changed later.
+"""
 type NativeTokenStatus {
-  unlocked: Boolean # If it's already unlocked
-  mintable: Boolean # User has permissions to mint new tokens
-  unlockable: Boolean # Token can be unlocked
+  """Whether the native token is unlocked."""
+  unlocked: Boolean
+  """Whether the user has permissions to mint new tokens."""
+  mintable: Boolean
+  """Whether the native token can be unlocked."""
+  unlockable: Boolean
 }
 
+"""
+Represents the status of a Colony.
+
+This contains important meta information about the Colony's token and other fundamental settings
+"""
 type ColonyStatus {
+  """Status information for the Colony's native token."""
   nativeToken: NativeTokenStatus
-  recovery: Boolean # if it's in recovery mode
+  """Whether the Colony is in recovery mode."""
+  recovery: Boolean
 }
 
-# Applies to both Colonies Tokens and Events, but not all fields are revlant to all
-# It does not apply to user accounts as they can live on all networks
+"""
+Represents metadata related to a blockchain event.
```

Comment:
(Genuine question, I don't know) Can we say it's only about a blockchain event? We use this type on tokens and colonies too